llvm/lib/CodeGen/MachineCopyPropagation.cpp

```diff
@@ -867,6 +867,12 @@ void MachineCopyPropagation::forwardUses(MachineInstr &MI) {
          make_range(Copy->getIterator(), std::next(MI.getIterator())))
       KMI.clearRegisterKills(CopySrcReg, TRI);
 
+    // Attempt to canonicalize/optimize the instruction now its arguments have
```

Comment:
The placement here is unsound.  You're in a loop which is operating over the instructions operands!

Reply:
Ouch yes, I'd meant to place it outside the loop. Thanks for catching - now fixed.